packages/url_launcher/url_launcher/README.md

```diff
@@ -43,55 +43,52 @@ See the example app for more complex examples.
 ## Configuration
 
 ### iOS
-Add any URL schemes passed to `canLaunchUrl` as `LSApplicationQueriesSchemes` entries in your Info.plist file.
+Add any URL schemes passed to `canLaunchUrl` as `LSApplicationQueriesSchemes`
+entries in your Info.plist file, otherwise it will return false.
 
 Example:
-```
+```xml
 <key>LSApplicationQueriesSchemes</key>
 <array>
-  <string>https</string>
-  <string>http</string>
+  <string>sms</string>
+  <string>tel</string>
 </array>
 ```
 
 See [`-[UIApplication canOpenURL:]`](https://developer.apple.com/documentation/uikit/uiapplication/1622952-canopenurl) for more details.
 
 ### Android
 
-Starting from API 30 Android requires package visibility configuration in your
-`AndroidManifest.xml` otherwise `canLaunchUrl` will return `false`. A `<queries>`
+Add any URL schemes passed to `canLaunchUrl` as `<queries>` entries in your
+`AndroidManifest.xml`, otherwise it will return false in most cases starting
+on Android 11 (API 30) or higher. A `<queries>`
 element must be added to your manifest as a child of the root element.
 
-The snippet below shows an example for an application that uses `https`, `tel`,
-and `mailto` URLs with `url_launcher`. See
-[the Android documentation](https://developer.android.com/training/package-visibility/use-cases)
-for examples of other queries.
-
+Example:
 ``` xml
 <queries>
-  <!-- If your app opens https URLs -->
-  <intent>
-    <action android:name="android.intent.action.VIEW" />
-    <data android:scheme="https" />
-  </intent>
-  <!-- If your app makes calls -->
+  <!-- If your app checks for call support -->
   <intent>
     <action android:name="android.intent.action.DIAL" />
```

Comment:
@GaryQian Are these actually right? They were added by a community-submitted PR that was based on comments from plugin clients in issue reports, but looking at them again after having more familiarity with the Android implementation it seems questionable.

Since the implementation of `canLaunchUrl` is always just `android.intent.action.VIEW` with a URL, shouldn't all of the queries be

```xml
  <intent>
    <action android:name="android.intent.action.VIEW" />
    <data android:scheme="{whatever scheme here}" />
  </intent>
```
?

Reply:
I've only been recently added as owner of this plugin, so I am actually unclear about this particular question.

Reply:
@blasten Are you familiar with the details of how query specification is supposed to work? Is my expectation that it should match the actual intent query we are going to make ([here](https://github.com/flutter/plugins/blob/main/packages/url_launcher/url_launcher_android/android/src/main/java/io/flutter/plugins/urllauncher/UrlLauncher.java#L41-L44)) correct?

Alternately, @GaryQian you could try experimenting locally to see if you can reproduce `canLaunchUrl` failing for various cases listed in this README section, and then if the query permission format I suggested above is all we need in each case to make it work? It could be a way to get to know this part of the plugin first-hand :)

Reply:
I agree: https://github.com/flutter/plugins/blob/main/packages/url_launcher/url_launcher_android/android/src/main/java/io/flutter/plugins/urllauncher/UrlLauncher.java#L41

It looks like we always hardcode ACTION_VIEW, so this DIAL intent may not be appropriate.

Reply:
I've updated to always use `VIEW` then, matching our implementation. PTAL.